core/net.go

```diff
@@ -376,18 +378,22 @@ func (n *OpenBazaarNode) SendCancel(peerID, orderID string) error {
 	//try to get public key from order
 	order, _, _, _, _, _, err := n.Datastore.Purchases().GetByOrderId(orderID)
 	var kp *libp2p.PubKey
+	var pub []byte
 	if err != nil { //probably implies we can't find the order in the Datastore
 		kp = nil //instead SendOfflineMessage can try to get the key from the peerId
+		pub = order.BuyerOrder.BuyerID.Pubkeys.Identity
 	} else {
 		k, err := libp2p.UnmarshalPublicKey(order.GetVendorListings()[0].GetVendorID().GetPubkeys().Identity)
 		if err != nil {
 			return err
 		}
 		kp = &k
+		pub = order.VendorListings[0].VendorID.Pubkeys.Identity
 	}
 	err = n.Datastore.Messages().Put(
 		fmt.Sprintf("%s-%d", orderID, int(pb.Message_ORDER_CANCEL)),
-		orderID, pb.Message_ORDER_CANCEL, peerID, repo.Message{Msg: m})
+		orderID, pb.Message_ORDER_CANCEL, peerID, repo.Message{Msg: m},
+		"", 0, pub)
```

Comment:
What's the significance of putting a `0` timestamp versus `time.Now()`?

Reply:
The field received_at is set to 0 is for outbound messages and time.Now() is for inbound messages.